packages/lablgtk3-sourceview3/lablgtk3-sourceview3.3.0.beta7/opam

```diff
@@ -0,0 +1,37 @@
+opam-version: "2.0"
+
+synopsis: "OCaml interface to GTK+ gtksourceview library"
+description: """
+OCaml interface to GTK+3, gtksourceview3 library.
+
+See http://lablgtk.forge.ocamlcore.org/ for more information.
+"""
+
+maintainer: "garrigue@math.nagoya-u.ac.jp"
+authors: ["Jacques Garrigue et al., Nagoya University"]
+homepage: "https://github.com/garrigue/lablgtk"
+bug-reports: "https://github.com/garrigue/lablgtk/issues"
+dev-repo: "git+https://github.com/garrigue/lablgtk.git"
+doc: "https://garrigue.github.io/lablgtk/lablgtk3-sourceview3"
+license: "LGPL with linking exception"
+
+depends: [
+  "ocaml"                {         >= "4.05.0"    }
+  "dune"                 { build & >= "1.6.0"
+                                 & != "1.7.0"
+                                 & != "1.7.1"     } # Due to dune/dune#1833
+  "lablgtk3"             {         >= "3.0.beta5" }
```

Comment:
This error seems to indicate that a slightly stricter constraint is needed here with either `{= version}` or something else.
```
#=== ERROR while compiling lablgtk3-sourceview3.3.0.beta7 =====================#
# context              2.0.5 | linux/x86_64 | ocaml-base-compiler.4.09.0 | pinned(https://github.com/garrigue/lablgtk/releases/download/3.0.beta7/lablgtk3-3.0.beta7.tbz)
# path                 ~/.opam/4.09/.opam-switch/build/lablgtk3-sourceview3.3.0.beta7
# command              ~/.opam/4.09/bin/dune build -p lablgtk3-sourceview3 -j 2
# exit-code            1
# env-file             ~/.opam/log/lablgtk3-sourceview3-25-023584.env
# output-file          ~/.opam/log/lablgtk3-sourceview3-25-023584.out
### output ###
#  dune_config src-sourceview3/cflag-gtksourceview-3.0.sexp,src-sourceview3/clink-gtksourceview-3.0.sexp
# Warning: Package name "gtksourceview-3.0 >= 3.18" contains invalid characters. Use Pkg_config.query_expr to construct proper queries
#       ocamlc src-sourceview3/ml_gtksourceview3.o (exit 2)
# (cd _build/default/src-sourceview3 && /home/opam/.opam/4.09/bin/ocamlc.opt -g -I /home/opam/.opam/4.09/lib/cairo2 -I /home/opam/.opam/4.09/lib/lablgtk3 -I /home/opam/.opam/4.09/lib/ocaml/threads -ccopt -pthread -ccopt -I/usr/include/gtksourceview-3.0 -ccopt -I/usr/include/libxml2 -ccopt -I/usr/include/gtk-3.0 -ccopt -I/usr/include/at-spi2-atk/2.0 -ccopt -I/usr/include/at-spi-2.0 -ccopt -I/usr/include/dbus-1.0 -ccopt -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -ccopt -I/usr/include/gtk-3.0 -ccopt -I/usr/include/gio-unix-2.0 -ccopt -I/usr/include/cairo -ccopt -I/usr/include/libdrm -ccopt -I/usr/include/pango-1.0 -ccopt -I/usr/include/harfbuzz -ccopt -I/usr/include/pango-1.0 -ccopt -I/usr/include/fribidi -ccopt -I/usr/include/atk-1.0 -ccopt -I/usr/include/cairo -ccopt -I/usr/include/pixman-1 -ccopt -I/usr/include/freetype2 -ccopt -I/usr/include/libpng16 -ccopt -I/usr/include/gdk-pixbuf-2.0 -ccopt -I/usr/include/libmount -ccopt -I/usr/include/blkid -ccopt -I/usr/include/uuid -ccopt -I/usr/include/glib-2.0 -ccopt -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -ccopt -Wno-deprecated-declarations -ccopt -g -o ml_gtksourceview3.o ml_gtksourceview3.c)
# ml_gtksourceview3.c:34:10: fatal error: wrappers.h: No such file or directory
#  #include <wrappers.h>
#           ^~~~~~~~~~~~
# compilation terminated.
```

Reply:
> This error seems to indicate that a slightly stricter constraint is needed here with either `{= version}` or something else.

This problem is fixed in `lablgtk3` git, but manually. I'm not sure how the `version` field is supposed to be used; can we include it without adding a `version: NNN` top-level field to the opam file?

Reply:
the version field is optional (and usually unwanted/redundant in opam-repository). `version` inside constraints blocks will always point to the version of the package as defined by either a `version` field or a the name of the directory `packages/<package>/<package>.<version>/opam` in opam-repository.

Anyway, I can see that the constraint is set to `>= "3.0.beta6"` is it correct to use it here? I was mostly pointing out to `{= version}` as it is the simplest way to do it and doesn't require much thinking.

Reply:
Yup thanks, we will use `{= version}`, see https://github.com/garrigue/lablgtk/pull/99